### PR TITLE
feat: :sparkles: add YouTube playlist prop

### PR DESCRIPTION
**Pull request by spences10**
add additional props to allow for embedding of YouTube playlists.

### CODE REVIEW

**Review by PaulieScanlon — CHANGES_REQUESTED** [review]
Sorry mate, just merged the `nocookie` PR which now means you've got conflicts on this one.

**Comment by spences10** [comment]
Aight!

I hadn't merged this one because the Twitch one was still failing 🙃

I'll rebase an merge

File: docs/pages/1.introduction.stories.mdx [hunk 1]
```diff
@@ -47,6 +47,10 @@ Here's a pen, and some other blog post text
 
 <Box sx={{ height: '24px' }} />
 
+<PayWithStripe />
```
Comment (PaulieScanlon) [inline docs/pages/1.introduction.stories.mdx:50]: Is this a rebase issue. Did you mean to commit this file?

File: docs/pages/components/youtube.stories.mdx [hunk 2]
```diff
@@ -63,3 +71,21 @@ Then use it like this.
...
 <YouTube youTubeId="WUgvvPRH7Oc" />
 ```
+
+To embed a YouTube playlist using the `<YouTube />` component grab the `youTubeListId` from the URL.
+
+```
+https://www.youtube.com/watch?v=C8M-JhhYaBs&list=PL5SvzogSTpeH1Szqw4tPi9ZfgXDbY8GU-
+```
+
+The `youTubeListId` is this bit.
```
Comment (PaulieScanlon) [inline docs/pages/components/youtube.stories.mdx:81]: Does this new functionality refer to playlists? Wondering if the prop should be called `youTubePlaylistId`

File: docs/src/components/pay-with-stripe.js [hunk 3]
```diff
@@ -4,7 +4,7 @@ import { ThemeProvider, Heading, Input, Button, Grid, Flex, Box, Link, Text, Spi
 
 import theme from '../theme';
 
-import { StripeLogo } from './stripe-logo.js';
+import { StripeLogo } from './stripe-logo';
```
Comment (PaulieScanlon) [inline docs/src/components/pay-with-stripe.js:7]: Did you make these changes?

File: docs/src/components/stripe-logo.js [hunk 4]
```diff
@@ -4,77 +4,32 @@ import { Box } from 'theme-ui';
 export const StripeLogo = () => {
   return (
     <Box
+      as="span"
```
Comment (PaulieScanlon) [inline docs/src/components/stripe-logo.js:7]: Did you make these changes?

File: packages/mdx-embed/src/components/youtube/youtube.test.tsx [hunk 5]
```diff
@@ -20,6 +20,17 @@ describe('<YouTube />', () => {
     expect(screen.getByTestId('youtube')).toBeDefined();
   });
 
+  test('it renders the component when provided a listId', () => {
```
Comment (PaulieScanlon) [inline packages/mdx-embed/src/components/youtube/youtube.test.tsx:23]: if you do make the change to the prop you might also need to change the description for the test

File: packages/mdx-embed/src/components/youtube/youtube.tsx [hunk 6]
```diff
@@ -4,7 +4,9 @@ import { GeneralObserver } from '../general-observer';
 
 export interface IYouTubeProps {
   /** YouTube id */
-  youTubeId: string;
+  youTubeId?: string | undefined;
```
Comment (PaulieScanlon) [inline packages/mdx-embed/src/components/youtube/youtube.tsx:7]: This prop is optional so i don't think it'll need the `| undefined`

File: packages/mdx-embed/src/components/youtube/youtube.tsx [hunk 7]
```diff
@@ -30,6 +33,11 @@ export const YouTube: FunctionComponent<IYouTubeProps> = ({
 
   const startTime = tH + tM + s;
 
+  const baseUrl = 'https://www.youtube.com/embed/';
```
Comment (PaulieScanlon) [inline packages/mdx-embed/src/components/youtube/youtube.tsx:36]: A change has been merged that conditionally decides if the "no cookie" url should be used so you might have conflicts on this.

**Review by PaulieScanlon — CHANGES_REQUESTED** [review]

File: docs/src/components/pay-with-stripe.js (hunk 3 above)
Comment (spences10) [reply docs/src/components/pay-with-stripe.js:7]: not me sir

File: docs/pages/components/youtube.stories.mdx (hunk 2 above)
Comment (spences10) [reply docs/pages/components/youtube.stories.mdx:81]: Yeah, would be more descriptive, right

File: docs/pages/1.introduction.stories.mdx (hunk 1 above)
Comment (spences10) [reply docs/pages/1.introduction.stories.mdx:50]: This was a rebase, I didn't touch that yo!

File: docs/src/components/stripe-logo.js (hunk 4 above)
Comment (spences10) [reply docs/src/components/stripe-logo.js]: no, i did not

**Pull request merged**